File: src/main/scala/com/kapilkoju/autopos/trade/sale/repo/SaleRepo.scala
package com.kapilkoju.autopos.trade.sale.repo

import com.kapilkoju.autopos.catalog.domain.SaleStatus
import com.kapilkoju.autopos.kernel.service.AuditableRepository
import com.kapilkoju.autopos.trade.sale.domain.Sale
import org.springframework.data.domain.{Page, Pageable}

trait SaleRepo extends AuditableRepository[Sale] {

  def findByStatus(status: SaleStatus, pageable: Pageable): Page[Sale]

}

File: src/main/kotlin/com/kapilkoju/autopos/trade/sale/service/SaleRepo.kt
package com.kapilkoju.autopos.trade.sale.service

import com.kapilkoju.autopos.catalog.domain.SaleStatus
import com.kapilkoju.autopos.kernel.service.AuditableRepository
import com.kapilkoju.autopos.trade.sale.domain.Sale
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface SaleRepo : AuditableRepository<Sale> {

    fun findByStatusAndVat(status: SaleStatus, vat: Boolean, pageable: Pageable): Page<Sale>

}
